flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -179,4 +171,20 @@ public DynamicTableSource copy() {
   public String asSummaryString() {
     return "Iceberg table source";
   }
+
+  private static TableSchema projectSchema(TableSchema tableSchema, int[] projectedFields) {
```

Comment:
In this method, we only project the first level fields in the nested schema. I mean in theory we should project other deeper level fields for implementing the nested schema projection, right ? 

In my view,  if we treat the  `tableSchema` as a tree data structure, then the nested projection schema should be a sub tree which is pruned from the whole `tableSchema` tree, and the sub tree should have the same root as the original tree.

Reply:
I didn't use nested projection in the first place. Initially, I pull the data using the top-level projection(topProjectedFields mentioned above), and then prun the data based on projected field index(int[][]);

Reply:
That's why I've added `topProjectedFields` and `projectedFields`